### ISSUE TITLE
Apps not recognizing Google Play services - MicroG is installed

### ISSUE THREAD

**Issue by Jesse-V** [microg/android_packages_apps_GmsCore#225]
First off, thanks for this really great project! I'm excited to see a FOSS solution to the Google framework for Android.

I'm currently having an issue registering Signal. After I type in my phone number, it says "Signal relies on Google Play services, which is not supported by your device. Contact the manufacturer for assistance." Despite my best efforts with MicroG, I'm not able to get past this. I have made some progress, since it used to say that Google Play services was not available, but that message went away after I installed MicroG. Has anyone experienced this issue? Is something wrong with my setup? Tickets #135 and #198 are somewhat related, but they and [the wiki](https://github.com/microg/android_packages_apps_GmsCore/wiki/Helpful-Information) suggest that Signal should be working. Please advise.

Background:
I'm running CyanogenMod 13.0 on the OnePlus One. I first installed the patches from github.com/Lanchon/haystack to enable signature spoofing, which I enabled in Developer Options and it appears to be working successfully. I then downloaded com.google.android.gms-9258259.apk and com.google.android.gsf-8.apk from microg.org and installed them using adb. Next, I disabled CyanogenMod's PrivacyGuard on MicroG Services Core and MicroG Services Framework so that there weren't any permissions restrictions there. I then rebooted the device. I added a Google account and signed in just fine. Within MicroG Settings, I checked "Enable device checkin" and "Enable Google Cloud Messaging". The Self-Check shows checks for the following items: "System spoofs signature", "Play Services (GmsCore) Installed", "Play Store (GmsCore) has correct signature", "Services Framework (GSF) Installed", "Android Version Supported", and "UnifiedNlp is registered in system". I rebooted again and then used adb to install Signal. However, I then encountered the aforementioned error message when I attempted to register my number. By all rights everything should be in place, so I am not sure what the issue is.

**Comment by n76** [comment]
Did you try to re-install Signal after you setup MicroG?

I've got Signal installed and while I haven't used it much, it does seem to work with MicroG.

**Comment by Jesse-V** [comment]
@n76 yes, I have tried clearing app data, uninstalling the app, and then reinstalling it again. Same issue. I get the feeling that I did something out of order, but I'm not sure what. It's not clear what the "Enable device checkin" feature does or if I needed to enable that before or after adding the Google account.

**Comment by n76** [comment]
I am pretty sure you need to enable device check in and "Google Cloud Messaging" to have Signal work.

**Comment by Jesse-V** [comment]
@n76 I have done this. I enabled that after signing into Google. The Google-developed Play services aren't on my phone, only MicroG on top of CyanogenMod.

**Comment by n76** [comment]
Not sure which version of GmsCore the APK you downloaded is. . . I install mine through the MicroG F-Droid repository and it is showing version 0.2.4-22-gcb356d2. Not sure it makes a difference, but I am running a self-compiled AOSP v6.1 ROM rather than CyanogenMod.

It has been a while, but I am pretty sure that I setup the core messaging after I created my Google account. (Out of caution I set up a throwaway Google account a long time ago that is the one I use for this type of stuff.) And I am pretty sure I enabled all the core messaging stuff before installing Signal.

I'd show you a screenshot of the Signal app but it seems that Signal disables screen capture (which I was unaware of until just now).

**Comment by mar-v-in** [comment]
@Jesse-V are you on 0.2.4 release or latest preview (0.2.4-27)? Please try with preview release and report back. Also, if possible provide a logcat from the moment the message appears in Signal.

**Comment by Jesse-V** [comment]
@n76 Thanks for the tips. I wasn't able to get the F-Droid repo set up properly, so I'm downloading from microg.org and then installing over adb. I installed MicroG's gmsCore, installed the GSF component, signed into Google, then enabled cloud messaging. I then restarted my device, installed Signal, double-checked that MicroG was working, then tried to register Signal. Same issue.

@mar-v-in Thanks, I didn't see the drop-down for the preview releases. I uninstalled Core and GSF and then installed 0.2.4-27 and GSF again. The app seems to have really matured in the previous release, at least looking at the UI and the features in MicroG Settings. However, I am seeing the same issue. "Play Store (Phonesky) Installed", "Battery Optimization ignored", "System supports location provider", and the last two UnifiedNlp items are not checked, but everything else is.

It's worth noting that it's not just Signal. I recently installed another app (Duo Mobile) and it also didn't recognize that Google Play services is there.

**Comment by n76** [comment]
At what point does Duo Mobile fail for you? Out of curiosity I installed it (using Blankstore) and it did not give my any error messages up to the point where I had to either scan a bar code or manually enter some account information.

**Comment by Jesse-V** [comment]
@n76 I installed it via adb and I was able to re-add my company account, no problem. It just has a message banner saying "Duo Push may not work because Google Play Services is unavailable or out of date. Tap to update." and when I tap it, I get a nearly identical message to the error with Signal. This shows that it's not Signal's fault.

I haven't installed BlankStore or FakeStore, just the  Does this matter?

**Comment by Jesse-V** [comment]
OMG IT WORKS! I re-read the instructions on the wiki and realized that I hadn't installed BlankStore or FakeStore. I had ignored this step because it seemed optional in the wiki and because the official GMS/push worked just fine when I had disabled the Play Store originally. I uninstalled Signal, installed BlankStore, and then installed Signal, and it worked right away!

MicroG is a beautiful project, I'm so happy right now.
